theano/gpuarray/corr_gemm.c

```diff
@@ -213,24 +372,43 @@ int im2col(const size_t max_threads_dim,
   size_t n_blocks = (num_kernels + threads_per_block - 1) / threads_per_block;
   int err;
   GpuKernel *kernel;
+  // Here, we only accept non signed stride. So no need to compare to SADDR32_MIN
+  int use_int = (data_im_offset < ADDR32_MAX) &&
+    (stride_h < SADDR32_MAX) && (stride_w < SADDR32_MAX);
```

Comment:
Stride here is not the stride for an array, but the convolution stride.  Also it can't ever be negative due to being a size_t.  Finally I'm pretty sure this test isn't enough.  We have to make sure `kernel_h * kernel _w * height_col * width_col`, `stride_h * kernel_w * height_col * width_col * h_col` and  `stride_w * height_col * width_col * w_col` don't go over.  There may be other key expressions like this.